solc_select/solc_select.py

```diff
@@ -90,8 +137,23 @@ def switch_global_version(version):
 def valid_version(version):
     match = re.search(r"^(\d+).(\d+).(\d+)$", version)
 
+    url = f"https://binaries.soliditylang.org/{soliditylang_platform()}/list.json"
+    list_json = urllib.request.urlopen(url).read()
+    latest_release = json.loads(list_json)["latestRelease"]
+
```

Comment:
only the `if StrictVersion(version) > StrictVersion(latest_release)` branch depends on this, can be moved there

Reply:
Changed to validate regex first, then minimum version, then maximum version in d583428